src/clj/puppetlabs/http/client/async.clj

```diff
@@ -147,7 +178,8 @@
    * :query-params - used to set the query parameters of an http request"
   [opts :- common/RawUserRequestOptions
    callback :- common/ResponseCallbackFn
-   client :- HttpAsyncClient]
+   client :- HttpAsyncClient
+   metric-registry :- common/OptionalMetricRegistry]
```

Comment:
Ah yeah. I was about to ask "what's our API", but I guess everything in the "Public" section we consider our API? It seems not super likely to be used given that we have `make-request` now, but it's very easy for me to make this not an API change, so I'll do that.